src/components/EmojiPickerControl.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+
+import data from "@emoji-mart/data";
+import Picker from "@emoji-mart/react";
+import { Drawer, Popover } from "@mantine/core";
+import { useMediaQuery } from "@mantine/hooks";
+import {
+  RichTextEditorControl,
+  useRichTextEditorContext,
+} from "@mantine/tiptap";
+import { IconMoodSmile } from "@tabler/icons-react";
+
+import styles from "./EmojiPicker.module.css";
+
+import { EmojiData } from "#/types/emoji";
+
+const EmojiPickerControl = () => {
+  const { editor } = useRichTextEditorContext();
+  const isMobile = useMediaQuery("(max-width: 768px)");
```

Comment:
`max-width: 768px` は，$x \leq 768 \space \text{px}$ を意味します。

恐らく `isMobile` は，スマートフォンのような小さな画面サイズのことを想定していると捉えました。そうなると，細かい話，例えばスクリーンサイズが `768x1024` の iPad でスマートフォンに最適化されたレイアウトが表示されてしまいます。

```suggestion
  const isMobile = useMediaQuery("(max-width: 767px)");
```

Reply:
個人的には、少なくともこの実装においてiPadはモバイルと同じUIのほうが良いと考えました。タッチ操作が前提であるデバイスにおいてはPopoverよりDrawerのほうが操作しやすいと思います(Mattermostの場合はタブレットサイズだとDialogを採用しているため難しいところですが)。
この場合変数名は`isDesktop`などのほうがわかりやすいでしょうか？

(別件で、具体的にブレイクポイントとしてどの値を採用するかは共通化しておいたほうが良さそうですね)

Reply:
なるほど！　確かにそれならもっともだと思います。

---

> 別件で、具体的にブレイクポイントとしてどの値を採用するかは共通化しておいたほうが良さそうですね

こちらも同意です。ブレイクポイントによる表示の出し分けは，いずれ複数の component で行う可能性が大いにあります。それゆえ，ユーティリティー関数のような形で独立させておきたいですねー。